### PR TITLE
Fix #2427 by converting $remember to integer

### LINKED ISSUE #2427: Statement.php change breaks oc_authtoken insertion
### Steps to reproduce
1. Use Nextcloud server after 9b808c4
2. Try to log in on a computer without an auth token

### Expected behaviour
Statement.php's "remember" value should be an integer, either a 0 or a 1 depending on whether the session should be remembered. The "remember" column of the "oc_authtokens" table contains only integers.

### Actual behaviour
The remember value is either "true" or "false", not integers.  A database exception occurs as a result of trying to insert the word "true" or "false" into the "remember" column.

>SQLSTATE[22P02]: Invalid text representation: 7 ERROR: invalid input syntax for integer: "f"

Because the auth token can't be stored in the database, you can't log in.

### Server configuration
**Operating system**:
Gentoo Linux
**Web server:**
Apache
**Database:**
PostgreSQL
**PHP version:**
7.0.13
**Nextcloud version:** (see Nextcloud admin page)
11.0 beta 2
**Updated from an older Nextcloud/ownCloud or fresh install:**
Yes
**Where did you install Nextcloud from:**
Most recently, the web updater.

### PR DESCRIPTION
**Pull request by justin-sleep**
Fixes #2427

### CODE REVIEW

**Comment by rullzer** [comment]
@justin-sleep thanks a lot for reporting the bug and fixing it as well!

@ChristophWurst please have a look :)

File: lib/private/User/Session.php [hunk 1]
```diff
@@ -558,7 +558,7 @@ public function createSessionToken(IRequest $request, $uid, $loginName, $passwor
 		try {
 			$sessionId = $this->session->getId();
 			$pwd = $this->getPassword($password);
-			$this->tokenProvider->generateToken($sessionId, $uid, $loginName, $pwd, $name, IToken::TEMPORARY_TOKEN, $remember);
+			$this->tokenProvider->generateToken($sessionId, $uid, $loginName, $pwd, $name, IToken::TEMPORARY_TOKEN, +$remember);
```
Comment (LukasReschke) [inline lib/private/User/Session.php]: (int)$remember maybe?

Comment (justin-sleep) [inline lib/private/User/Session.php]: As I understand it from [PHP's page on arithmetic operators](http://php.net/manual/en/language.operators.arithmetic.php) isn't +$remember essentially the same? Or is it better to explicitly cast as int here?

Comment (blizzz) [inline lib/private/User/Session.php]: It's not the same. ``+$var`` also would convert to float, while ``(int)`` explicitly converts to int.

```
 php > $s3 = '12.3';
php > $a = +$s3;
php > var_dump($a);
php shell code:1:
double(12.3)
php > $a = (int)$s3;
php > var_dump($a);
php shell code:1:
int(12)
```

This should not apply here, and nevertheless it's better to be crisp and clear :) Also, imho ``(int)`` has a better readability.

Comment (rullzer) [inline lib/private/User/Session.php]: While they might be semanticly the same. I think what @LukasReschke means more is that explicitly casting with (int) makes it more readable.

File: lib/private/User/Session.php [hunk 2]
```diff
@@ -558,7 +558,7 @@ public function createSessionToken(IRequest $request, $uid, $loginName, $passwor
 		try {
 			$sessionId = $this->session->getId();
 			$pwd = $this->getPassword($password);
-			$this->tokenProvider->generateToken($sessionId, $uid, $loginName, $pwd, $name, IToken::TEMPORARY_TOKEN, $remember);
+			$this->tokenProvider->generateToken($sessionId, $uid, $loginName, $pwd, $name, IToken::TEMPORARY_TOKEN, (int)$remember);
```
Comment (nickvergessen) [inline lib/private/User/Session.php]: I think this should either be at the top of the chain (LoginController: https://github.com/nextcloud/server/blob/d75e35b75e9d70e511bee2c9fc830825363a4fd6/core/Controller/LoginController.php#L243 )
or the bottom (DefaultTokenProvider: https://github.com/nextcloud/server/blob/2183a1f3e6dddae2f49a0bd9ad30e1670f2d9d15/lib/private/Authentication/Token/DefaultTokenProvider.php#L94-L94)

but not in the middle?

Comment (nickvergessen) [inline lib/private/User/Session.php]: And I think top is the way to go, because also this method states that it should be called with an integer only.

Comment (ChristophWurst) [inline lib/private/User/Session.php]: Yup sounds good!

**Comment by MorrisJobke** [comment]
@justin-sleep Could I ask you to do the changes as requested above? Then this is good to get in :rocket:

**Comment by justin-sleep** [comment]
Look good?

**Comment by nickvergessen** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Comment by MorrisJobke** [comment]
Thanks a lot @justin-sleep - feel free to join our IRC channel #nextcloud-dev on freenode or the forums :) Maybe also check out some other starter issues in https://github.com/nextcloud/server/issues?q=is%3Aissue+is%3Aopen+label%3A%22starter+issue%22 😉 Have a nice weekend :)

**Pull request merged**